dll/cpl/sysdm/general.c

```diff
@@ -417,14 +419,330 @@ static VOID SetProcSpeed(HWND hwnd, HKEY hKey, LPTSTR Value, UINT uID)
+    }
+    StringCchCopyW(pwsFind, lenFind, L" ");
+    StringCchCatW(pwsFind, lenFind, pwFind);
+    StringCchCatW(pwsFind, lenFind, L" ");
+
+    if (!(pwsBuf = wcsistr(pwsStr, pwsFind)))
+    {
+        goto freeFind;
+    }
+
+    pwsReplace = HeapAlloc(GetProcessHeap(), 0, lenReplace * sizeof(WCHAR));
+    if (!pwsReplace)
+    {
+        goto freeFind;
+    }
+    StringCchCopyW(pwsReplace, lenReplace, L" ");
+    StringCchCatW(pwsReplace, lenReplace, pwReplace);
+    StringCchCatW(pwsReplace, lenReplace, L" ");
+
+    do
+    {
+        // replace substring
+        memmove(pwsBuf, pwsReplace, (lenReplace - 1) * sizeof(WCHAR));
+        // shift characters
+        memmove(pwsBuf + lenReplace - (wcslen(pwReplace) > 0 ? 1 : 2), pwsBuf + lenFind - 1, (lenStr - lenFind - (pwsBuf - pwsStr) + 1) * sizeof(WCHAR));
+    }
+    while ((pwsBuf = wcsistr(pwsStr, pwsFind)) != NULL);
+
+    TrimDmiStringW(pwsStr);
+    StringCchCopyW(pwStr, wcslen(pwStr), pwsStr);
+
+    HeapFree(GetProcessHeap(), 0, pwsReplace);
+freeFind:
+    HeapFree(GetProcessHeap(), 0, pwsFind);
+freeStr:
+    HeapFree(GetProcessHeap(), 0, pwsStr);
+}
+
+static BOOL GetSystemName(PWSTR pBuf, SIZE_T cchBuf)
+{
+    PVOID SMBiosBuf;
+    PCHAR DmiStrings[ID_STRINGS_MAX] = { 0 };
+    WCHAR ven[512], dev[512];
+    UINT i;
+
+    SMBiosBuf = LoadSMBiosData(DmiStrings);
+    if (!SMBiosBuf)
+    {
+        return FALSE;
+    }
+
+    GetSMBiosStringW(DmiStrings[SYS_VENDOR], ven, _countof(ven), TRUE);
+    GetSMBiosStringW(DmiStrings[SYS_PRODUCT], dev, _countof(dev), TRUE);
+
+    if (wcslen(dev) == 0 ||
+        !wcscmp(dev, ven) ||
+        !wcscmp(dev, L"All Series") || // some ASUS boards
+        !wcscmp(ven, L"To be filled by O.E.M.") || // various boards
+        !wcscmp(dev, L"To be filled by O.E.M.") || // various boards
+        !wcscmp(ven, L"To Be Filled By O.E.M.") || // various boards
+        !wcscmp(dev, L"To Be Filled By O.E.M.") || // various boards
+        !wcscmp(ven, L"System manufacturer") || // some ASUS boards
+        !wcscmp(dev, L"System Product Name") || // some ASUS boards
+        !wcscmp(dev, L"DEPO Computers") || // various boards
+        !wcscmp(ven, L"DEPO Computers") || // various boards
+        (!wcscmp(ven, L"KRAFTWAY") && !wcscmp(dev, L"GEG")) || // various boards
+        !wcscmp(dev, L"Aquarius Server") || // some ASUS server boards
+        !wcscmp(dev, L"Super Server")) // some Supermicro server boards
+    {
+        // system strings are unusable, use board strings
+        GetSMBiosStringW(DmiStrings[BOARD_VENDOR], ven, _countof(ven), TRUE);
+        GetSMBiosStringW(DmiStrings[BOARD_NAME], dev, _countof(dev), TRUE);
+
+        if (wcslen(ven) == 0 && wcslen(dev) == 0)
+        {
+            // board strings are empty, use BIOS vendor string
+            GetSMBiosStringW(DmiStrings[BIOS_VENDOR], ven, _countof(ven), TRUE);
+        }
+    }
+
+    // make vendor strings shorter
+    wcsrep(ven, L"ASUSTeK", L"ASUS");
```

Comment:
Does windows do this?

Reply:
I don't know whether Windows does this at all, clearly a new feature :)